### PR TITLE
Fix #4111

### LINKED ISSUE #4111: Android: "Provided world path doesn't exist" if chosing singleplayer on first run
Steps to reproduce:

If you remove/rename the Minetest folder from your sd card, start minetest as a "first run", and immediately tap on the singleplayer button, you will get an "Provided world path doesn't exist" error.

If you change tabs (and change them back), or restart minetest, you can play singleplayer. Still, the bug is bad for everyone first trying minetest, they will think singleplayer is completely broken.

I've tested the 0.4.13 build from f-droid and couldn't reproduce. This also means that this bug is no dupe of #2971. @ShadowNinja can reproduce it as well.

### PR DESCRIPTION
**Pull request by Rui914**
I'm sorry about my mistake.

### CODE REVIEW

File: builtin/mainmenu/init.lua [hunk 1]
```diff
@@ -77,13 +77,28 @@ local function init_globals()
...
 	if PLATFORM == "Android" then
 		local world_list = core.get_worlds()
-		local world_index = table.indexof(world_list, "singleplayerworld")
-
-		if world_index == -1 then
+		local world_index
+
+		local found_singleplayerworld = false
+		for i, v in ipairs(world_list) do
+			if world.name == "singleplayerworld" then
```
Comment (rubenwardy) [inline builtin/mainmenu/init.lua]: Isn't this meant to be v?

Comment (rubenwardy) [inline builtin/mainmenu/init.lua]: Or even better, change v to world in the for loop

**Comment by Wayward1** [comment]
Currently this crashes upon starting for me; will continue to test

**Comment by Wayward1** [comment]
adb logcat *:W returns this: http://paste.ubuntu.com/16376248/

**Comment by Zeno-** [comment]
Merging shortly with v/world error fixed

**Comment by Zeno-** [comment]
Fixed error and pushed:  9d3d43f

File: builtin/mainmenu/init.lua (hunk 1 above)
Comment (sofar) [inline builtin/mainmenu/init.lua]: It reads a bit funny, he only wants the first entry in the list.

You can write it a lot more readable like this:

```
if world.name == "singleplayerworld" then
        world_index = next(world_list, nil)
        found_singleplayerworld = true
end
```

**Comment by est31** [comment]
@Rui914 Thanks alot for the fix! This was really fast.

**Pull request closed without merging**